LibreNMS/OS/Ciscowlc.php

```diff
@@ -120,15 +121,13 @@ public function pollOS(): void
             ]));
 
             /** @var AccessPoint $db_ap */
-            if ($db_ap = $db_aps->get($ap->getCompositeKey())) {
-                $db_aps->forget($ap->getCompositeKey());
-                $ap = $db_ap->fill($ap->getAttributes());
+            if (! $db_aps->get($ap->getCompositeKey())) {
+                // Previously seen AP was not present during the poll, mark it as deleted
+                $ap->deleted = 1;
             }
-
             $ap->save(); // persist ap
         }
 
-        $db_aps->each->delete(); // delete those not removed
```

Comment:
Instead of these changes, just save valid ap ids and then delete all not in that list.

Reply:
Thought of this approach initially but there is a catch - the WLC will return only the APs that are present during the poll. If we would like to see which previously seen APs are missing (== offline) we would need to somehow mark them instead of removing them. This is why I am using the softdelete in the new wireless module.

Reply:
I think I will try to finish the new module instead of fixing the remaining issues in the old one.